### PR TITLE
Added tests and fix n+1 problems

### PR DESCRIPTION
**Pull request by markotuya0**
At this juncture, [alphayowakarindi](https://github.com/alphayowakarindi), [Pierre](https://github.com/Piercel2022) and I implemented the following:

Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.

Use Capybara to write integration tests for each view in the project.

User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.

I can see the user's username.

I can see the number of posts the user has written.

I can see a post's title.

I can see some of the post's body.

I can see the first comments on a post.

I can see how many comments a post has.

I can see how many likes a post has.

I can see a section for pagination if there are more posts than fit on the view.

When I click on a post, it redirects me to that post's show page.

I can see the post's title.

I can see who wrote the post.

I can see how many comments it has.

I can see how many likes it has.

I can see the post body.

I can see the username of each commentor.

I can see the comment each commentor left.

### CODE REVIEW

File: spec/views/posts/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  it 'shows first comments of the post' do
+    expect(page).to have_content('My first awesome comment straight from Kenya')
+  end
+
+  it 'shows how many comments a post has' do
+    expect(page).to have_content(1)
+  end
+
```
Comment (omar-muhamad) [inline spec/views/posts/index_spec.rb]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=see%20the%20user%27s%20username.-,I%20can%20see%20the%20number%20of%20posts%20the%20user%20has%20written.,-I%20can%20see%20a), Please add a Test that asserts you can see the number of posts the user has written.
- [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20a%20post%27s%20title.), Please add a Test that asserts you can see a post's title.
- [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20some%20of%20the%20post%27s%20body.), Please add a Test that asserts you can see some of the post's body.

File: spec/views/posts/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    Post.create(title: 'My third post', text: 'Body text of the post', author_id: @user.id)
+    Post.create(title: 'My fourth post', text: 'Body text of the post', author_id: @user.id)
+    Comment.create(text: 'My first awesome comment straight from Kenya', author_id: @user.id, post_id: @post_one.id)
+
+    visit user_post_path(@user, @post_one.id)
+  end
+
+  it 'renders who wrote the post ' do
+    expect(page.html).to include(@user.name)
+  end
```
Comment (omar-muhamad) [inline spec/views/posts/show_spec.rb]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20post%27s%20title.), Please add a Test that asserts you can see the post's title.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi Team,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/views/posts/index_spec.rb (hunk 1 above)
Comment (alphayowakarindi) [reply spec/views/posts/index_spec.rb]: Hi @Omar-Muhamad, thank you for taking the time to leave this excellent review. The changes above have been applied. Cheers

File: spec/views/posts/show_spec.rb (hunk 2 above)
Comment (markotuya0) [reply spec/views/posts/show_spec.rb]: Hi @Omar-Muhamad, thank you for taking the time to leave this excellent review. The changes above have been applied. Cheers

File: spec/views/posts/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  it 'shows number of posts written by agiven user' do
+    expect(page).to have_content(4)
+  end
+
+  it 'shows post title of the user' do
+    expect(page).to have_content('My first post')
+  end
+
+  it 'shows post title of the user' do
```
Comment (danmainah) [inline spec/views/posts/index_spec.rb]: - [ ] @team great work implementing the changes kindly change the title of this last text, to indicate the test is used to show some section of the post body.

File: spec/views/posts/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    expect(page).to have_content(@post_one.text)
+  end
+
+  it 'renders username of each commentor ' do
+    expect(page.html).to include(@user.name)
+  end
+
+  it 'renders comment of each commentor ' do
+    expect(page).to have_content('My first awesome comment straight from Kenya')
+  end
```
Comment (danmainah) [inline spec/views/posts/show_spec.rb:31]: - [ ] @team great work on your test however as the previous reviewer requested kindly add a test that shows you can see the post's title. 
image

Comment (danmainah) [reply spec/views/posts/show_spec.rb:31]: - [ ] @team I also noted some tests are not passing when I run the test, kindly ensure all the tests pass.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @team, 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] Kindly ensure to use the most recent readme file for documentation, [template link](https://github.com/microverseinc/readme-template)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
Hi team, 

Thanks for making the requested changes 👏 

### Highlights ✨ 

- Pull request 👌 
- Git flow ✅ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**